Microsoft.Azure.Cosmos/src/Query/v3Query/QueryDefinition.cs

```
@@ -89,6 +89,58 @@ public QueryDefinition WithParameter(string name, object value)
             return this;
         }
 
+        /// <summary>
+        /// Checks for equality of two QueryDefinitions ignoring order of sqlParameters
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns>Boolean representing the equality</returns>
+        public bool Equals(QueryDefinition other)
+        {
+            if (this.QueryText != other.QueryText)
+            {
+                return false;
+            }
+
+            if (this.SqlParameters.Count != other.SqlParameters.Count)
```

This is probably cheaper to compare than the preceding comparison, so you'll fail faster if you move it up

---

Good point, thanks